local/php/envs_test.go

```diff
@@ -182,6 +182,138 @@ func (s *PHPFPMSuite) TestGenerateEnv(c *C) {
 				"SCRIPT_NAME":     "/index.php",
 			},
 		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory",
+			expected: map[string]string{
+				"PATH_INFO":       "",
+				"REQUEST_URI":     "/subdirectory",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php",
+				"REQUEST_URI":     "/subdirectory/unknown.php",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php/",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php/",
+				"REQUEST_URI":     "/subdirectory/unknown.php/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/index.php/foo",
+			expected: map[string]string{
+				"PATH_INFO":       "/foo",
+				"REQUEST_URI":     "/subdirectory/index.php/foo",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "///subdirectory",
+			expected: map[string]string{
+				"PATH_INFO":       "///subdirectory",
+				"REQUEST_URI":     "///subdirectory",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/index.php",
+				"SCRIPT_NAME":     "/index.php",
+			},
```

Comment:
because `/subdirectory/index.php` exists I guess one would expect this to match it